File: fib/fib.c
#include <stdio.h>

void fib(int n, long *buf) {
  for (int i = 2; i < n; i++)
    buf[i] = buf[i - 1] + buf[i - 2];
}

int main() {
  long buf[50] = {0, 1};
  fib(50, buf);
  for (int i = 1; i < 50; i++)
    printf("%ld\n", buf[i]);
  return 0;
}

File: fib/Fib.java
import java.math.BigInteger;
import java.util.Arrays;
import java.util.ArrayList;

public class Fib {

    public static ArrayList<BigInteger> fib(int n) {
	ArrayList<BigInteger> buf = new ArrayList<>(Arrays.asList(BigInteger.ZERO, BigInteger.ONE));
	for (int i = 2; i < n; i++)
	    buf.add(buf.get(i - 1).add(buf.get(i - 2)));
	return buf;
    }
    
    public static void main(String[] args) {
	for (BigInteger n: fib(100))
	    System.out.println(n);
    }
    
}
